src/test/java/org/llorllale/cactoos/matchers/MatcherEnvelopeTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.llorllale.cactoos.matchers;
+
+import java.util.Random;
+import org.cactoos.BiProc;
+import org.cactoos.Func;
+import org.cactoos.Proc;
+import org.cactoos.text.Randomized;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Test case for {@link MatcherEnvelope}.
+ *
+ * @since 1.0.0
+ */
+public final class MatcherEnvelopeTest {
+
+    /**
+     * A rule for handling an exception.
+     */
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
```

Comment:
@scristalli There's a `Throws` matcher, which I think could be used instead of this rule.

Reply:
@vzurauskas I tried this
```
@Test
public void decoratesDescribeTo() {
    new Assertion<>(
        "must throw an AssertionError",
        () -> {
            new Assertion<>(
                "must delegate describeTo to encapsulated matcher",
                MatcherEnvelopeTest.TEST_INTEGER + 1,
                new MatcherEnvelopeChild<>(new EncapsulatedTestMatcher())
            ).affirm(); 
            return null;
        },
        new Throws<>(AssertionError.class)
    ).affirm();
}
```

The problem is that `affirm()` returns `void`, while the tested lambda has to return `Object`. I see no way to get rid of the nasty `return` statement, even without `null` it wouldn't be much prettier. I suggest we leave the tests as they are for now (the other tests in cactoos-matchers use the same rule as well); maybe we could face this issue separately?

Reply:
@vzurauskas I found this use of `Assertion` in [IsTrueTest](https://github.com/llorllale/cactoos-matchers/blob/master/src/test/java/org/llorllale/cactoos/matchers/IsTrueTest.java). There, `return true` is used. I followed the same approach and adapted the tests here in commit 313d2d5.